### PR TITLE
[Release 5.1 hotfix] copr: avoid panic when response is large

### PR DESCRIPTION
**Pull request by ekexium**
### What is changed and how it works?

Issue Number: ref #9012 

What's Changed:

```commit-message
Return an error instead of panicking when gRPC response size is too large
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: src/coprocessor/dag/mod.rs [hunk 1]
```diff
@@ -128,7 +128,19 @@ fn handle_qe_response(
     match result {
         Ok(sel_resp) => {
             let mut resp = Response::default();
-            resp.set_data(box_try!(sel_resp.write_to_bytes()));
+            let data = sel_resp.write_to_bytes();
+            if data.is_err() {
+                info!(
+                    "handle copr select resp";
```
Comment (cfzjywxk) [inline src/coprocessor/dag/mod.rs:134]: Maybe print the error in the log too.

**Comment by purelind** [comment]
@ekexium Need comments /`run-build` to trigger ci `idc-jenkins-ci-tikv/build`

**Review by you06 — APPROVED** [review]
For other reviewers: the panic is avoided by protobuf upgrading.

**Pull request merged**